applications/vanilla/controllers/api/CommentsApiController.php

```diff
@@ -233,34 +233,50 @@ public function index(array $query) {
                 'minimum' => 1,
                 'maximum' => 100
             ],
+            'insertUserID:i?' => 'Filter by author.',
             'after:dt?' => 'Get only comments after this date.',
             'expand:b?' => [
                 'description' => 'Expand associated records.',
                 'default' => false
-            ]
-        ], 'in')->setDescription('List comments.');
+            ],
+            'lookup:s?' => 'The field used to lookup comments.'
+        ], 'in')->requireOneOf(['discussionID', 'insertUserID'])->setDescription('List comments.');
         $out = $this->schema([':a' => $this->commentSchema()], 'out');
 
         $query = $in->validate($query);
-        $discussion = $this->discussionByID($query['discussionID']);
-        list($offset, $limit) = offsetLimit("p{$query['page']}", $query['limit']);
-        $this->discussionModel->categoryPermission('Vanilla.Discussions.View', $discussion['CategoryID']);
 
-        // Build up the where clause.
-        $where = ['discussionID' => $query['discussionID'], 'joinUsers' => false];
+        // Lookup by discussion or by user?
+        if (array_key_exists('discussionID', $query)) {
+            $discussion = $this->discussionByID($query['discussionID']);
+            list($offset, $limit) = offsetLimit("p{$query['page']}", $query['limit']);
+            $this->discussionModel->categoryPermission('Vanilla.Discussions.View', $discussion['CategoryID']);
 
-        if (isset($query['after'])) {
-            $where['dateInserted >'] = $query['after'];
+            // Build up the where clause.
+            $where = ['discussionID' => $query['discussionID'], 'joinUsers' => false];
```

Comment:
I did a double check as an FYI and only table names are case sensitive in MySQL.
https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html